spanner/api/Spanner.Samples/LogCommitStatsAsync.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright 2021 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_get_commit_stats]
+
+using Google.Cloud.Spanner.Data;
+using Google.Cloud.Spanner.V1;
+using Google.Cloud.Spanner.V1.Internal.Logging;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+public class LogCommitStatsAsyncSample
+{
+    public async Task<long> LogCommitStatsAsync(string projectId, string instanceId, string databaseId)
+    {
+        // Commit statistics are logged at level Info by the default logger.
+        // This sample uses a custom logger to access the commit statistics.
+        // See https://googleapis.github.io/google-cloud-dotnet/docs/Google.Cloud.Spanner.Data/logging.html
+        // for more information on how to use loggers.
+        var logger = new CommitStatsSampleLogger();
+        var options = new SessionPoolOptions();
+        var poolManager = SessionPoolManager.Create(options, logger);
+        var connectionStringBuilder = new SpannerConnectionStringBuilder
+        {
+            ConnectionString = $"Data Source=projects/{projectId}/instances/{instanceId}/databases/{databaseId}",
+            // Set LogCommitStats to true to enable logging commit statistics for all transactions on the connection.
+            // LogCommitStats can also be enabled/disabled for individual Spanner transactions.
+            LogCommitStats = true,
+            SessionPoolManager = poolManager,
+        };
+
+        using var connection = new SpannerConnection(connectionStringBuilder);
+        await connection.OpenAsync();
+
+        using var cmd = connection.CreateDmlCommand("INSERT Singers (SingerId, FirstName, LastName) VALUES (110, 'Virginia', 'Watson')");
+        var rowCount = await cmd.ExecuteNonQueryAsync();
+        var mutationCount = logger._lastCommitResponse.CommitStats.MutationCount;
+
+        Console.WriteLine($"{rowCount} row(s) inserted...");
+        Console.WriteLine($"{mutationCount} mutation(s) in transaction...");
+
+        return mutationCount;
+    }
+
+    /// <summary>
+    /// Sample logger that keeps a reference to the last seen commit response.
+    /// Use the default logger if you only want to log the commit stats.
+    /// </summary>
+    public class CommitStatsSampleLogger : Logger
```

Comment:
Why do we wan to show a custom logger? In particular this one that seem to add nothing (except maybe for our own testing purposes). I think the idea was that clients shouldn't have to change their code and if the only sample we have includes a custom logger, then I wonder if we are sending the wrong signal?

Reply:
Ah this was because in other languages, the user is able to read the value of the mutation count in code. Whereas, they can't do that by default in C#. So we thought that it would be good to show the user how to read the value out from the logger if possible in case they need to make a decision based on that.

Reply:
Hmmm, still feels more like an "advance" scenerio for a second sample than the simple usage that I would show in the only sample. But I'm totally fine with leaving that decision to you.